### PR TITLE
[1LP][RFR] Chargeback resource allocation : Add support for other providers

### PR DESCRIPTION
**Pull request by nachandr**
{{pytest: cfme/tests/intelligence/chargeback/test_resource_allocation.py --use-provider complete --long-running}}

### CODE REVIEW

File: cfme/tests/intelligence/chargeback/test_resource_allocation.py [hunk 1]
```diff
@@ -297,8 +413,7 @@ def generic_test_resource_alloc(resource_alloc, chargeback_report_custom, column
             if 'GB' in groups[column] and column == 'Memory Allocated over Time Period':
                 allocated_resource = float(allocated_resource) * math.pow(2, -10)
             resource_from_report = groups[column].replace('MB', '').replace('GB', '')
-            soft_assert(allocated_resource - DEVIATION <=
```
Comment (mshriver) [inline cfme/tests/intelligence/chargeback/test_resource_allocation.py]: Looks like removing the deviation might be causing test failures in run 23882

File: cfme/tests/intelligence/chargeback/test_resource_allocation.py [hunk 2]
```diff
@@ -129,23 +131,137 @@ def verify_vm_uptime(appliance, provider):
...
+            appliance.db.client.session.query(rollups.id)
+            .join(ems, rollups.parent_ems_id == ems.id)
+            .filter(rollups.capture_interval_name == 'hourly', rollups.resource_name == vm_name,
+            ems.name == provider.name, rollups.timestamp >= date.today())
+        )
+
+    for record in appliance.db.client.session.query(rollups).filter(
+            rollups.id.in_(result.subquery())):
+        if (record.cpu_usagemhz_rate_average or
+           record.cpu_usage_rate_average or
```
Comment (rlbabyuk) [inline cfme/tests/intelligence/chargeback/test_resource_allocation.py]: Required: please check indentation here

File: cfme/tests/intelligence/chargeback/test_resource_allocation.py [hunk 3]
```diff
@@ -34,7 +36,7 @@
     pytest.mark.meta(blockers=[BZ(1511099, forced_streams=['5.9', '5.8'],
                                   unblock=lambda provider: not provider.one_of(GCEProvider)),
                                ]),
-    pytest.mark.provider([SCVMMProvider], scope='module',
+    pytest.mark.provider([InfraProvider], scope='module', selector=ONE_PER_TYPE,
```
Comment (rlbabyuk) [inline cfme/tests/intelligence/chargeback/test_resource_allocation.py]: Question: Please explain this replace. can we have both of them in these tests

File: cfme/tests/intelligence/chargeback/test_resource_allocation.py [hunk 4]
```diff
@@ -129,23 +131,137 @@ def verify_vm_uptime(appliance, provider):
...
             message='Waiting for VM to be up for at least one hour')
 
-    vm = appliance.rest_api.collections.vms.get(name=vm_name)
-    vm.reload(attributes=['allocated_disk_storage', 'cpu_total_cores', 'ram_size'])
+        vm = appliance.rest_api.collections.vms.get(name=vm_name)
+        vm.reload(attributes=['allocated_disk_storage', 'cpu_total_cores', 'ram_size'])
+        # By default,chargeback rates for storage are defined in this form: 0.01 USD/GB
+        # Hence,convert storage used in Bytes to GB
+        return {"storage_alloc": int(vm.allocated_disk_storage * math.pow(2, -30)),
+            "memory_alloc": int(vm.ram_size),
```
Comment (rlbabyuk) [inline cfme/tests/intelligence/chargeback/test_resource_allocation.py]: Required: check indentation

File: cfme/tests/intelligence/chargeback/test_resource_allocation.py (hunk 3 above)
Comment (nachandr) [reply cfme/tests/intelligence/chargeback/test_resource_allocation.py]: I added support for SCVMM through another PR.
I'm adding support for the other providers through this PR.
I have also provided explanation in the module itself.

File: cfme/tests/intelligence/chargeback/test_resource_allocation.py (hunk 4 above)
Comment (nachandr) [reply cfme/tests/intelligence/chargeback/test_resource_allocation.py]: done

File: cfme/tests/intelligence/chargeback/test_resource_allocation.py (hunk 2 above)
Comment (nachandr) [reply cfme/tests/intelligence/chargeback/test_resource_allocation.py]: done

**Comment by dajoRH** [comment]
Lint report for commit 87bcd18f0ef2553c1018ff98251594f15281a140:

`cfme/tests/intelligence/chargeback/test_resource_allocation.py`:
- :red_circle: Line 438:46: **F821** *undefined name 'RESOURCE_DEVIATION'*
- :red_circle: Line 439:69: **F821** *undefined name 'RESOURCE_DEVIATION'*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

**Pull request merged**